### PR TITLE
Define our set of GitHub labels for open source SDK repositories

### PR DESCRIPTION
**Pull request by QuintinWillison**
See [this Slack thread (internal)](https://ably-real-time.slack.com/archives/C03JDBVM5MY/p1665741282123009) for more context.

To speed things up for reviewers, while this pull request remains open, the rendered output from this change can be viewed on the underlying branch [here](https://github.com/ably/engineering/blob/sdk-github-labels/sdk/github.md#labels).

### CODE REVIEW

File: sdk/github.md
Comment (deanna-lad) [inline sdk/github.md]: Some features create breaking changes so should we amend the phrasing so that it says "Implies a need to release related changes in a 'minor' or 'major' version bump?

**Review by deanna-lad — APPROVED** [review]

File: sdk/github.md
Comment (QuintinWillison) [reply sdk/github.md]: Thanks, this is a valid observation. If a feature is breaking then my expectation is that it would have _both_ the `enhancement` and `breaking` labels... this is something I'm expanding on for my next pull request which enhances the guidance around releases, but in the meantime I think I can add some explanatory wording to this table on this point. I'll push up a commit shortly to do that...

Comment (QuintinWillison) [reply sdk/github.md]: Wording added: https://github.com/ably/engineering/pull/44/commits/b3cc90f638f0795ecec3a0cfe4d02b3118f68029

File: sdk/github.md
Comment (ikbalkaya) [inline sdk/github.md:215]: This description looks too generic to me. I think it should be something like. "Breaks existing API contract with users"

File: sdk/github.md
Comment (ikbalkaya) [inline sdk/github.md:216]: I am not sure whether this label would only be applicable for the cases where user experience is affected. Shouldn't any internal code quality improvements (that are invisiible to our users) be part of this?

File: sdk/github.md
Comment (ikbalkaya) [inline sdk/github.md:218]: I am not sure if that should be the case, but what if we do a work that improves some internal working of a visible function (eg - speed improvement)? Isn't that also an enhancement? And is it a breaking change?

**Review by ikbalkaya — CHANGES_REQUESTED** [review]
Some questions and recommendations

File: sdk/github.md
Comment (QuintinWillison) [reply sdk/github.md:216]: That's what was intended to be meant by "in our codebase". Could you suggest better wording that makes it clearer?

File: sdk/github.md
Comment (QuintinWillison) [reply sdk/github.md:218]: What you describe doesn't sound like a breaking change, but could be an `enhancement` depending on how impactful it is and whether we feel customers need to know about it. I'm not sure what could be added to this table to enhance that understanding though, so open to suggested changes or additions to the wording here. 🤔

File: sdk/github.md
Comment (QuintinWillison) [reply sdk/github.md]: Corrected: https://github.com/ably/engineering/pull/44/commits/49fae80e4c3d76dfd480b92ab3730f3f86faf217

File: sdk/github.md
Comment (QuintinWillison) [reply sdk/github.md]: Does this help?: https://github.com/ably/engineering/pull/44/commits/a2f1ce07c6b840d2902d45e70ee1ea71d09ea174

**Review by Peter-Maguire — APPROVED** [review]

File: sdk/github.md
Comment (ikbalkaya) [reply sdk/github.md:218]: I would personally prefer a different label for a new feature (such as new-feature) and separate it from enhancement. But if we have to keep them on the same label, I think description might be  something like "New feature or functionality enhancement" and then the implication for the release would only be for cases when new features are added.

File: sdk/github.md
Comment (ikbalkaya) [reply sdk/github.md]: Ah of course. I think my mind quickly jumped to the user as we use 'developer experience' on the context of our SDK users 🤦

File: sdk/github.md
Comment (owenpearson) [inline sdk/github.md:220]:
```suggestion
| `failing-test` | ![ff8888](https://img.shields.io/badge/-ff8888-ff8888) | Where a test is failing either locally or in CI. Perhaps flakey, wrong or bug. |
```
I would remove this since a test can be flaky without being badly written, for example an integration test which tests integration with a flaky service.

File: sdk/github.md
Comment (ikbalkaya) [reply sdk/github.md:218]: I will resolve this conversation and leave it to you whether you want to change anything here as we have agreed on this in the office

**Review by ikbalkaya — APPROVED** [review]

File: sdk/github.md
Comment (owenpearson) [inline sdk/github.md:218]: I think this should be under a separate label for `feature` - as it stands none of these labels would be appropriate for a performance improvement or any other kind of improvement to an existing feature (both of which I think should fall under `enhancement`).

Comment (owenpearson) [reply sdk/github.md:218]: (I know @ikbalkaya already suggested the same but he resolved the conversation and I personally would like to see it addressed before I approve)

**Review by owenpearson — CHANGES_REQUESTED** [review]

File: sdk/github.md
Comment (QuintinWillison) [reply sdk/github.md]: I've taken aboard your suggestion, with slightly different wording: https://github.com/ably/engineering/pull/44/commits/34456b2a587f61429ba0fab21e181761ab79bee9

File: sdk/github.md
Comment (owenpearson) [reply sdk/github.md]: resolving as [34456b2](https://github.com/ably/engineering/pull/44/commits/34456b2a587f61429ba0fab21e181761ab79bee9) has addressed my point

**Pull request merged**
